### DISCUSSION TITLE
Can't get whitelist or blacklist to work

### DISCUSSION THREAD

**Discussion by mayakerostasia** [jertel/elastalert2#877] [Q&A] [answered]
I've followed the documentation and looked through the oneOf schema and I'm pretty sure I have everything that is required but it's still not working for me

My goal is to alert (once a day -- but I still have to figure the frequency part of it - still learning the elastalert2 system) but all objects in my index that have Non-zero missing agents

The es cluster required connection information has been passed in environment variables where possible. I have other rules taken from the examples working.

The rule yaml
```
# Rule
name: Nonzero Missing_Agents
type: whitelist
index: missing_agents

# Rule_Type Required
compare_key: missing_agents_response.total_count 
ignore_null: false
whitelist: 
- 0

# Alert
alert:
  - "debug"
```

The error is
```
elastalert.util.EAException: Invalid Rule file: rules/show_nonzero_missing_agents.yaml
{'name': 'Nonzero Missing_Agents', 'type': 'whitelist', 'index': 'missing_agents', 'compare_key': 'missing_agents_response.total_count', 'ignore_null': False, 'whitelist': [0], 'alert': ['debug'], 'rule_file': 'rules/show_nonzero_missing_agents.yaml'} is not valid under any of the given schemas
Failed validating 'oneOf' in schema:
```

I tried to include all information I felt was pertinent, please let me know if you need more and I will provide.

I apologize if this is excessively basic.

**Comment by jertel** [comment, marked as answer]
The `whitelist:` property requires strings, yet a numeric `0` was provided. Try `"0"`.
